src/extension.ts

```diff
@@ -26,6 +27,14 @@ export function activate(context: vscode.ExtensionContext) {
 
   vscode.workspace.onDidChangeTextDocument((e: vscode.TextDocumentChangeEvent) => {
     NoteParser.updateCacheFor(e.document.uri.fsPath);
+    
+    const shouldSuggest = e.contentChanges.some((change) => {
```

Comment:
Can you explain what this code does in some inline comments? Looks like something around only triggered suggest if the cursors is at the end of a word not in the middle?

Reply:
Yep, the situation I want to handle here is kind of tricky.

Suppose I want to type a Chinese word "概念" (i.e. concept) by using an IME, I'd have to type "gai", select "概", then type "nian", and select "念".

When I test this process on MacOS, surprisingly, vscode didn't show me any useful suggestions. So I added some log output at the beginning of [provideCompletionItems](https://github.com/kortina/vscode-markdown-notes/blob/68356ffcd1f4cf6c1026bda79b072bf7d3d03550/src/MarkdownFileCompletionItemProvider.ts#L19), and noticed that only the following words triggered the completion provider:
```
g
ga
gai
概n
概ni
概nia
概nian
```

"概" and "概念" are missing from the list, which suggests that when I finish typing a single Chinese character, vscode didn't invoke the completion provider for me.

However, because the complete correct word is "概念", only the prefix "概" or "概念" can suggest the correct word, the completion provider need to be triggered at that moment.

Thus I spent a little more time investigating the conditions for triggering suggest, and realized that vscode only triggers a suggest for an *insertion*. Selecting a correct Chinese character is not an insertion, is an replacement. Thus I need a little ad-hoc logic to trigger a suggest when an replacement happened inside `[[]]`.

Reply:
I guess I'd better write a block of comments here to briefly explain what happened, and maybe leave a link to this PR so the other developers can understand the reason behind this ad-hoc logic.

Reply:
Oh, interesting.

Do you foresee this changing behavior when offering completions for other languages? Would be great to understand more examples.

I will also checkout the code on my machine and mess around to see if there is any noticeable change in behavior in cases other than the one you mentioned.

Reply:
I wonder if -- to be on the safe side in case there is unexpected behavior -- we put this behavior behind a new configuration, `vscodeMarkdownNotes.triggerSuggestOnReplacement`? That way in case there is some case we did not think of  where this new ad hoc trigger is not desirable, people  can turn it off.

And perhaps we default the setting to `false`? Unless you are very confident this is unlikely to cause odd behaviors, in which case I could be convinced we default it to `true`

Lmk what you think. 

This PR is looking pretty good, btw, so I plan to merge once  a few more of these comments / questions are addressed.

Reply:
As far as I know, Chinese and Japanese use IME to select characters, some languages like French or Vietnamese may use IME to select accents, all these languages should get benefits from this fix.

For the other users, I think the only way they can trigger this ad-hoc logic is copy-pasting something into a `[[]]`. I personally feel okay to trigger a suggest in this case, because I'd want to have some suggestions when I'm changing a wiki-link anyway.

So, I like the idea about adding a new configuration, `vscodeMarkdownNotes.triggerSuggestOnReplacement` LGTM, and I suggest to default it to `true`. LMK if you find any interesting corner case that may trigger this ad-hoc thing in an annoying way, let's see what can be done for it.

Reply:
Added here: https://github.com/kortina/vscode-markdown-notes/pull/76

Reply:
Would you mind reviewing / testing the setting to make sure I didn't break?